alf/utils/resnet50.py

```diff
@@ -0,0 +1,91 @@
+# Copyright (c) 2019 Horizon Robotics. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+from tensorflow.keras import layers
+
+
+def conv_block(input_tensor,
```

Comment:
Will it be better if this function is a class that derives from keras.Model?

Reply:
Right, let's have an uniform way to construct model: create the layers or Variables in `__init__` (of keras.Model or tf.Module) and use them later in call() or other functions. It's easier to reason about the owner of the Variables.

Reply:
okay, i'll modify it

Reply:
add a class `BottleneckBlock` inherited from `tf.keras.models.Model`